File: exercism/go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("strings must have equal length")
	}
	res := 0
	for i := range a {
		if a[i] != b[i] {
			res++
		}
	}
	return res, nil
}

File: exercism/c/hamming/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs) {
  int res = 0;
  for (; *lhs && *rhs; lhs++, rhs++) {
    if (*lhs != *rhs)
      res++;
  }
  if (*lhs || *rhs)
    return -1;

  return res;
}
